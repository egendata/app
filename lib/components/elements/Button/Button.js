import React from 'react'
import { TouchableOpacity } from 'react-native'
import styled from '../../../theme'
import { Icon } from '../Icon/Icon'
import { BaseButton, BaseButtonText } from './BaseButton'

export const StyledPrimaryButton = styled(BaseButton)`
  background-color: ${({ theme, red }) =>
    red ? theme.colors.deny : theme.colors.primary};
`

export const PrimaryButtonText = styled(BaseButtonText)`
  color: ${({ theme }) => theme.colors.white};
`

export const PrimaryButton = ({ children, icon, style, red, ...props }) => {
  return (
    <TouchableOpacity style={{ flexDirection: 'row', ...style }} {...props}>
      <StyledPrimaryButton
        red={red}
        style={{
          shadowColor: 'lightgrey',
          shadowOffset: { width: 0, height: 6 },
          shadowOpacity: 0.5,
          shadowRadius: 6,
          elevation: 2,
        }}
      >
        {icon ? (
          <Icon
            name={icon.name || ''}
            width={icon.width || 24}
            height={icon.height || 24}
            fill={icon.fill || 'white'}
          />
        ) : null}
        <PrimaryButtonText>{children}</PrimaryButtonText>
      </StyledPrimaryButton>
    </TouchableOpacity>
  )
}

export const SecondaryButton = ({ children, ...props }) => (
  <TouchableOpacity {...props}>
    <BaseButton
      style={{
        shadowColor: '#B1ACEC',
        shadowRadius: 14,
        shadowOpacity: 0.2,
        shadowOffset: {
          width: 0,
          height: 8,
        },
        marginBottom: 32,
      }}
    >
      <BaseButtonText>{children}</BaseButtonText>
    </BaseButton>
  </TouchableOpacity>
)
