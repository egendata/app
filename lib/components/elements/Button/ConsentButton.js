import React from 'react'
import { TouchableOpacity, View, Text } from 'react-native'
import styled, { theme as globalTheme } from '../../../theme'
import { Icon } from '../Icon/Icon'

const StyledConsentButton = styled(View)`
  align-items: center;
  background-color: ${({ theme }) => theme.colors.white};
  flex-direction: row;
  justify-content: center;
`

const StyledConsentText = styled(Text)`
  color: ${({ theme }) => theme.colors.primary};
  font-family: ${({ theme }) => theme.typography.bold};
  font-size: ${({ theme }) => theme.fontSize.regular};
  padding: 28px 16px;
  text-align: center;
`
export const ConsentButtonWrap = styled(View)`
  align-items: center;
  background-color: ${({ theme }) => theme.colors.white};
  border-top-color: ${({ theme }) => theme.colors.border};
  border-top-width: 1px;
  flex-direction: row;
  height: 96px;
  justify-content: space-evenly;
`

export const AcceptConsentButton = ({ children, ...props }) => (
  <TouchableOpacity {...props}>
    <StyledConsentButton>
      <Icon name="check" fill={globalTheme.colors.accept} width={24} />
      <StyledConsentText>{children}</StyledConsentText>
    </StyledConsentButton>
  </TouchableOpacity>
)

export const DenyConsentButton = ({ children, ...props }) => (
  <TouchableOpacity {...props}>
    <StyledConsentButton>
      <Icon name="ban" fill={globalTheme.colors.deny} width={24} />
      <StyledConsentText>{children}</StyledConsentText>
    </StyledConsentButton>
  </TouchableOpacity>
)
