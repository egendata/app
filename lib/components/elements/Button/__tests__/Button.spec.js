import React from 'react'
import { fireEvent, render, shallow } from 'react-native-testing-library'
import MockedProvider from '../../../../utils/test-utils'
import { PrimaryButton } from '../Button'

describe('elements/Button', () => {
  it('should render', () => {
    const { output } = shallow(<PrimaryButton>PrimaryButton</PrimaryButton>)

    expect(output).toMatchSnapshot()
  })

  it('should call onClick-handler', () => {
    const handleClick = jest.fn()

    const { getByText } = render(
      <MockedProvider>
        <PrimaryButton onPress={handleClick}>PrimaryButton</PrimaryButton>
      </MockedProvider>,
    )

    fireEvent.press(getByText(/primarybutton/i))

    expect(handleClick).toHaveBeenCalledTimes(1)
  })
})
