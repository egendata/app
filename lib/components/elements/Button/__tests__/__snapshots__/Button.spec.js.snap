// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`elements/Button should render 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  style={
    Object {
      "flexDirection": "row",
    }
  }
>
  <ForwardRef
    style={
      Object {
        "elevation": 6,
        "shadowColor": "grey",
        "shadowOffset": Object {
          "height": 6,
          "width": 0,
        },
        "shadowOpacity": 0.2,
        "shadowRadius": 6,
      }
    }
  >
    <ForwardRef>
      PrimaryButton
    </ForwardRef>
  </ForwardRef>
</TouchableOpacity>
`;
